client/components/search/index.jsx

```diff
@@ -271,19 +280,26 @@ const Search = React.createClass( {
 			spellCheck: 'false'
 		};
 
-		searchClass = classNames( this.props.additionalClasses, {
+		const searchClass = classNames( this.props.additionalClasses, this.props.dir, {
 			'is-expanded-to-container': this.props.fitsContainer,
 			'is-open': isOpenUnpinnedOrQueried,
 			'is-searching': this.props.searching,
-			rtl: this.props.dir === 'rtl',
-			ltr: this.props.dir === 'ltr',
+			'no-close-button': this.props.hideClose,
+			'has-focus': this.state.hasFocus,
 			search: true
 		} );
 
+		const gridIconClass = classNames( 'search-open__icon', this.props.dir );
```

Comment:
Since there is another `gridIconClass` in `this.closeButton()` -- what about renaming this one to `searchIconClass`, and the other one to `closeIconClass`?